app/src/main/java/de/westnordost/streetcomplete/data/logs/LogsTable.kt

```diff
@@ -0,0 +1,25 @@
+package de.westnordost.streetcomplete.data.logs
+
+object LogsTable {
+    const val NAME = "logs"
+
+    object Columns {
+        const val ID = "id"
+        const val LEVEL = "level"
+        const val TAG = "tag"
+        const val MESSAGE = "message"
+        const val ERROR = "error"
+        const val TIMESTAMP = "timestamp"
+    }
+
+    const val CREATE = """
+        CREATE TABLE $NAME (
+            ${Columns.ID} int PRIMARY KEY,
```

Comment:
Removed the primary key as suggested.

Regarding an additional index on the timestamp: My reasoning was that this table will be very much write-heavy with very sporadic reads (few users will sporadically use the log viewer to troubleshoot bugs). This means that slowing frequent writes that will affect every user might not be worth a slightly faster filtering in the log viewer. I have not run any benchmarks to prove this, though. If you think an extra overhead for writes is not an issue, then I can add the index.

Reply:
A sound reasoning. You are right about that it is not accessed frequently. On the other hand, accessing any range of logs within the table without such an index requires SQLite to iterate the whole table each time. The table could be quite large, as you've upped the time limit to 2 weeks now. But to be honest, I have no idea how many log lines to expect in two weeks of usage.
I also don't have an idea how big the performance overhead of an index would be on insertion. However, a primary key is also an index and (pretty sure) implemented in the same way. In other words, any table with a primary key (pretty much every table) will have the same overhead - so I expect that this is somewhat streamlined. Insertion into a B-Tree shouldn't be a particularly costly operation.

So, gut feeling, I'd go with an index. But I'll merge this PR either way, so without further information on insertion overhead vs access overhead, I leave this up to you.

Reply:
Added an index after all. Might go back to benchmarking this after there's some real-world usage data (e.g. number of log messages generated per day), but right now it would all be premature optimization probably :)